### PR TITLE
feat(parquet): complete canonical INT96 interoperability

### PR DESCRIPTION
**Pull request by ibgreen**
## Goals

- Close the stable Parquet logical-type gap on both TypeScript read and write paths.
- Make legacy INT96 interoperability safe and explicit for Arrow and object-row users.
- Consolidate the remaining INT96 work into one reviewable follow-up on the latest `master`.

## Changes

- Add canonical INT96 writer support using Parquet's Julian-day plus nanoseconds-of-day layout.
- Make `parquet.int96AsTimestamp` select the canonical INT96 physical schema for Arrow `timestamp-nanosecond` fields.
- Emit canonical INT96 page-index bounds when page indexes are enabled, enabling selective timestamp scans.
- Preserve exact epoch nanoseconds with `bigint`, including timestamps before the Unix epoch.
- Reject malformed values, invalid nanoseconds-of-day fields, unsupported Julian days, and values outside signed 64-bit Arrow timestamp bounds.
- Keep raw legacy INT96 compatibility available when canonical timestamp mode is not requested.
- Document the writer option and update the Parquet support matrix.
- Extend focused Vitest coverage with canonical encode/decode and boundary cases.

## Validation

- `yarn lint fix`
- `yarn test-headless modules/parquet/test/parquet-int96.spec.ts modules/parquet/test/parquetjs/codec-plain.spec.ts` (24 tests passed)
- `yarn build` (passed; emitted only existing bundle warnings)

This PR starts from the latest `master`, where the reader-side INT96, page-index, encryption, and scan foundations are already landed. Proposed ALP, PFOR, and VECTOR encodings remain experimental and are intentionally outside this change.

### CODE REVIEW

File: modules/parquet/src/parquet-js-writer.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ type ParquetJSWriterEncoderOptions = {
   writeStatistics?: boolean | Record<string, boolean>;
   /** Declares row-group sort keys using top-level or dotted nested leaf names. */
   sortingColumns?: readonly ParquetSortingColumnOption[];
+  /** Encodes INT96 input values as epoch nanoseconds using the canonical Julian-day layout. */
+  int96AsTimestamp?: boolean;
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/parquet/src/parquet-js-writer.ts:60]: Fixed in 93c3656b58. The writer option now reaches schema conversion, and timestamp-nanosecond fields map to physical INT96 only when `parquet.int96AsTimestamp` is enabled. Added a writer-level schema test to cover the public path.

File: modules/parquet/src/parquetjs/codecs/plain.ts [hunk 2]
```diff
@@ -154,15 +154,33 @@ function decodeValues_INT64(
...
   const buf = new Uint8Array(12 * values.length);
   for (let i = 0; i < values.length; i++) {
-    writeInt64LE(buf, values[i], i * 12);
-    writeUInt32LE(buf, values[i] >= 0 ? 0 : 0xffffffff, i * 12 + 8);
+    if (!asTimestamp) {
+      writeInt64LE(buf, values[i], i * 12);
+      writeUInt32LE(buf, values[i] >= 0 ? 0 : 0xffffffff, i * 12 + 8);
+      continue;
+    }
+    const epochNanoseconds = BigInt(values[i]);
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/parquet/src/parquetjs/codecs/plain.ts:165]: Fixed in 93c3656b58. Canonical INT96 encoding now rejects epoch nanoseconds outside the signed 64-bit range before Julian-day conversion, with focused regression coverage.
